HW11.py

```diff
@@ -0,0 +1,129 @@
+"""
+Homework 11. Bank Card operations.
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and as minimum  one staticmethod,
+# two or more getters/setters, __str_ magic method
+# do not forget about block if __name__ == '__main__':
+# and code there to check your class written logic.
+"""
+
+import re
+from datetime import datetime as dt
+from random import SystemRandom as Sr
+
+rand = Sr()
+
+
+class CardOperations:
+    """Card operations."""
+
+    CVV_DIGITS = 3
+    card_type = 'Standard'
+
+    def __init__(self, cvv, card_num, end_date, surname, name, pin, money=0):
+        """Has params."""
+        self.__CVV = cvv if self.check_cvv(cvv) else str(rand.randint(1, 999)).zfill(3)
+        self.__PIN = pin
+        self.Name = name
+        self.Surname = surname
+        self.EndDate = end_date
+        self.__Card_num = card_num if card_num else str(rand.randint(1, 9999999999999999)).zfill(12)
+        self.Money = float(money)
```

Comment:
Money or balance better to convert to protected attribute. For example when we want to check balance then always pin required for this . Users are not allowed to access this attribute directly